### ISSUE TITLE
Exception caught: Error: no decoy PSMs were provided.

### ISSUE THREAD

**Issue by nukaemon** [bzhanglab/DeepRescore#7]
Dear developers

In run_percolator process, I got an error `Exception caught: Error: no decoy PSMs were provided`.
I'm not sure if this is related with --decoy_prefix option, but I'm currently setting XXX_ because I don't know exactly what should be given to this option.

<details>
<summary>log message from nextflow</summary>

N E X T F L O W  ~  version 20.10.0
Launching `/home/centos/DeepRescore/DeepRescore.nf` [goofy_hopper] - revision: 86e61de38a
executor >  local (10)
[14/0f974e] process > xml2mzid (d2)                       [100%] 1 of 1 ✔
[a7/3c6ef2] process > calc_basic_features_xt (d2)         [100%] 1 of 1 ✔
[70/8e7f0f] process > pga_fdr_control (d2)                [100%] 1 of 1 ✔
[00/3a5478] process > generate_train_prediction_data (d2) [100%] 1 of 1 ✔
[3b/b226be] process > run_pdeep2 (d2)                     [100%] 1 of 1 ✔
[1c/c4d0de] process > process_pDeep2_results (d2)         [100%] 1 of 1 ✔
[ac/b6b3cf] process > train_autoRT (d2)                   [100%] 1 of 1 ✔
[2c/f2da0f] process > predicte_autoRT (d2)                [100%] 1 of 1 ✔
[ea/2a9ade] process > generate_percolator_input (d2)      [100%] 1 of 1 ✔
[d2/be169a] process > run_percolator (d2)                 [  0%] 0 of 1
[-        ] process > generate_pdv_input                  -
Error executing process > 'run_percolator (d2)'

Caused by:
  Process `run_percolator (d2)` terminated with an error exit status (1)

Command executed:

  percolator -r d2_pep.txt -m d2_psms.txt format.pin

Command exit status:
  1

Command output:
  (empty)

Command error:
  Percolator version 3.04.0, Build Date Sep 19 2019 15:17:34
  Copyright (c) 2006-9 University of Washington. All rights reserved.
executor >  local (10)
[14/0f974e] process > xml2mzid (d2)                       [100%] 1 of 1 ✔
[a7/3c6ef2] process > calc_basic_features_xt (d2)         [100%] 1 of 1 ✔
[70/8e7f0f] process > pga_fdr_control (d2)                [100%] 1 of 1 ✔
[00/3a5478] process > generate_train_prediction_data (d2) [100%] 1 of 1 ✔
[3b/b226be] process > run_pdeep2 (d2)                     [100%] 1 of 1 ✔
[1c/c4d0de] process > process_pDeep2_results (d2)         [100%] 1 of 1 ✔
[ac/b6b3cf] process > train_autoRT (d2)                   [100%] 1 of 1 ✔
[2c/f2da0f] process > predicte_autoRT (d2)                [100%] 1 of 1 ✔
[ea/2a9ade] process > generate_percolator_input (d2)      [100%] 1 of 1 ✔
[d2/be169a] process > run_percolator (d2)                 [100%] 1 of 1, failed: 1 ✘
[-        ] process > generate_pdv_input                  -
Error executing process > 'run_percolator (d2)'

Caused by:
  Process `run_percolator (d2)` terminated with an error exit status (1)

Command executed:

  percolator -r d2_pep.txt -m d2_psms.txt format.pin

Command exit status:
  1

Command output:
  (empty)

Command error:
  Percolator version 3.04.0, Build Date Sep 19 2019 15:17:34
  Copyright (c) 2006-9 University of Washington. All rights reserved.
  Written by Lukas Käll (lukall@u.washington.edu) in the
  Department of Genome Sciences at the University of Washington.
  Issued command:
  percolator -r d2_pep.txt -m d2_psms.txt format.pin
  Started Wed Sep  8 07:59:18 2021
   on 58ed86d6c6f8
  Hyperparameters: selectionFdr=0.01, Cpos=0, Cneg=0, maxNiter=10
  Reading tab-delimited input from datafile format.pin
  
  WARNING: Tab delimited input does not contain ScanNr column,
           scan numbers will be assigned automatically.
  
  Features:
  Mass_Error Abs_Mass_Error Ln_Total_Intensity Match_Ions_Intensity Rel_Match_Ions_Intensity Max_Match_Ion_Intensity X\!Tandem:expect X\!Tandem:hyperscore Charge error similarity 
  Found 1165 PSMs
  Concatenated search input detected, skipping both target-decoy competition and mix-max.
  Train/test set contains 1165 positives and 0 negatives, size ratio=inf and pi0=1
  Exception caught: Error: no decoy PSMs were provided.

Work dir:
  /home/centos/Work/210908.F24Z019E/deep_rescore/work/d2/be169ae511d224425606f95f999982

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

</detail>

**Comment by nukaemon** [comment]
I just realized that my identification file not including any decoy sequences.
I recreated my identification file by providing a fasta including decoy sequence to X!tandem, then the error went away.

I used [DecoyPYrat](https://github.com/wtsi-proteomics/DecoyPYrat) to generate a decoy sequence fasta.
It can specify prefix for decoy entries and actually sets "XXX_" prefix by default.
